YubiKit/YubiKit/Sessions/Shared/APDU/OATH/YKFOATHRenameAPDU.m

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018-2019 Yubico AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "YKFOATHRenameAPDU.h"
+#import "YKFKeyOATHRenameRequest.h"
+#import "YKFAPDUCommandInstruction.h"
+#import "YKFAssert.h"
+#import "YKFNSMutableDataAdditions.h"
+#import "YKFOATHCredential+Private.h"
+
+static const UInt8 YKFOATHRenameAPDUNameTag = 0x71;
+
+@implementation YKFOATHRenameAPDU
+
+- (instancetype)initWithRequest:(nonnull YKFKeyOATHRenameRequest *)request {
+    YKFAssertAbortInit(request);
+    
+    NSMutableData *rawRequest = [[NSMutableData alloc] init];
+    
+    // Current name
+    NSString *name = request.credential.key;
+    NSData *nameData = [name dataUsingEncoding:NSUTF8StringEncoding];
+    [rawRequest ykf_appendEntryWithTag:YKFOATHRenameAPDUNameTag data:nameData];
+    
+    // New name
+    YKFOATHCredential *credential = request.credential;
+    credential.issuer = request.issuer;
+    credential.account = request.account;
+    NSString *newName = credential.key;
```

Comment:
You missing period. If period is not default value - it's part of the key. In case of this renaming it will reset period to default value.

Reply:
We're simply changing the issuer and account of the same credential we used to calculate the key on line 32. Period, if set, will be the same as previously.

Reply:
Got it...I missed the point that you're not creating new credential, but reusing the passed one.